modules/widgets/my-community.php

```diff
@@ -0,0 +1,240 @@
+		$instance['include_followers']  = (bool) $new_instance['include_followers'];
+		$instance['include_commenters'] = (bool) $new_instance['include_commenters'];
+
+		delete_transient( "$this->id-{$instance['number']}" . (int) $instance['include_likers'] . (int) $instance['include_followers'] . (int) $instance['include_commenters'] . '-widget-data' );
+
+		return $instance;
+	}
+
+	/**
+	 * Front-end display of widget.
+	 *
+	 * @see WP_Widget::widget()
+	 *
+	 * @param array $args     Widget arguments.
+	 * @param array $instance Saved values from database.
+	 */
+	function widget( $args, $instance ) {
+		$title = isset( $instance['title' ] ) ? $instance['title'] : false;
+
+		if ( false === $title ) {
+			$title = $this->default_title;
+		}
+
+		$title = apply_filters( 'widget_title', $title );
+
+		echo $args['before_widget'];
+
+		if ( ! empty( $title ) ) {
+			echo $args['before_title'] . $title . $args['after_title'];
+		}
+
+		$transient_name = "$this->id-{$instance['number']}" . (int) $instance['include_likers'] . (int) $instance['include_followers'] . (int) $instance['include_commenters'] . '-widget-data';
+
+		$my_community = get_transient( $transient_name );
+
+		if ( empty( $my_community ) ) {
+			$my_community = $this->get_community( $instance );
+
+			set_transient( $transient_name, $my_community, self::$expiration );
+		}
+
+		echo $my_community;
+
+		echo $args['after_widget'];
+	}
+
+	/**
+	 * Initiate request and render the response.
+	 *
+	 * @since 4.0
+	 *
+	 * @param array $query
+	 *
+	 * @return string
+	 */
+	function get_community( $query ) {
+		$members = $this->fetch_remote_community( $query );
+
+		if ( ! empty( $members ) ) {
+
+			$my_community = '<div class="widgets-multi-column-grid"><ul>';
+
+			foreach ( $members as $member ) {
+				$my_community .= sprintf( '<li><a href="%s" %s><img alt="" src="%s" class="avatar avatar-240" height="48" width="48" originals="240" scale="1" /></a></li>',
+					$member->profile_URL,
+					empty( $member->name ) ? '' : 'title="' . $member->name . '"',
+					$member->avatar_URL
+				);
+			}
+
+			$my_community .= '</ul></div>';
+
+		} else {
+			if ( current_user_can( 'edit_theme_options' ) ) {
+				$my_community = '<p>' . wp_kses( sprintf( __( 'There are no users to display in this <a href="%1$s">My Community widget</a>. <a href="%2$s">Want more traffic?</a>', 'jetpack' ),
+						admin_url( 'widgets.php' ),
+						'http://en.support.wordpress.com/getting-more-site-traffic/'
+					), array( 'a' => array( 'href' => true ) ) ) . '</p>';
+			} else {
+				$my_community = '<p>' . esc_html__( "I'm just starting out; leave me a comment or a like :)", 'jetpack' ) . '</p>';
+			}
+		}
+
+		return $my_community;
+	}
+
+	/**
+	 * Request community members to WordPress.com endpoint.
+	 *
+	 * @since 4.0
+	 *
+	 * @param $query
+	 *
+	 * @return array
+	 */
+	function fetch_remote_community( $query ) {
+		$jetpack_blog_id = Jetpack_Options::get_option( 'id' );
```

Comment:
Since the Widget module doesn't require a connection to WordPress.com, that could be an issue. I think we should check if the blog is connected to wpcom before to do anything.